Correct bugs in variable names usage

lavirott opened the pull request.

lavirott committed:
Correct bugs in variable names usage

lavirott committed:
Update Readme.md to correspond to script modifications

lavirott committed:
Update script fr Raspbian 9 and Debian 9.6
Replace isc-dhcp-server config by the dhcpcd default one

sebastianzillessen commented on Readme.md:
```diff
@@ -1,34 +1,32 @@
-chmod +x install.sh
-./install.sh
+git clone https://github.com/lavirott/wifi-ap-pi3.git
+cd wifi-ap-pi3
```
how did you come to the new folder, when I check it out I get `WiFi-AP-Raspberry-Pi-3`

lavirott replied:
I rebased code depending on my repository (which I renaming to avoid the use of uppercase letters). This was not supposed to be included in the pull request.

lavirott replied:
OK, I made a rollback with the new pull request (it was an error).

sebastianzillessen commented on Readme.md:
```diff
@@ -1,34 +1,32 @@
-3. IP_RANGE (default: 192.168.1, produces IP of Pi with 192.168.1.1)
+1. SSID (default: RaspberryPi )
+2. PASSPHRASE(default: raspberry)
+3. IP_RANGE (default: 192.168.222, produces IP of Pi with 192.168.222.1)
```
why changing the Ip addresses and the SSID/Password?

sebastianzillessen commented on install.sh:
```diff
@@ -1,11 +1,31 @@
 WIFI="${5:-wlan0}"
 
+pause() {
+#	read -p "Appuyer sur une touche pour continuer ..." var
```
No comments in other languages. Please remove.

lavirott replied:
I will correct that, of course.

lavirott replied:
The sentence is now in English with the new pull request. I also added a comment in the script to clarify.

sebastianzillessen commented on install.sh:
```diff
@@ -14,106 +34,173 @@ echo " IP-Address: $IP_RANGE.1"
-# check wlan0 available
-
-if ! ifconfig -a | grep "$WIFI"; then
+echo
```
Why `echo`?

lavirott replied:
During the begin function, the display is without new line (to permit the done to be displayed on the same line). In cas you want to display this error, I choose to display it on a single line. So I added an echo for adding newline. In case you modify the begin function and don't do an echo -n, the error will be more easy to find because, you will have a blank line to separe it from the normal flow (it could or should also be send to error flow, I should correct it).

sebastianzillessen commented on install.sh:
```diff
@@ -14,106 +34,173 @@ echo " IP-Address: $IP_RANGE.1"
+
+# Check if INC (eth0) and WIFI (wlan0) are available
+begin "Test if interfaces exist..."
+ifconfig -a | grep "$WIFI" > /dev/null
```
don't see the advantage in storing this in `/dev/null` and then getting it via `$?`. Can you explain me whats the advantage here?

lavirott replied:
This avoid to display the standard result of the command on the terminal. Only errors will be displayed.

lavirott replied:
And I prefered to use the $? (which contains the error code of the last executed command) for a better reading of the code.

lavirott replied:
As I use a redirection to avoid displaying info in console, I had to split it into two lines.

sebastianzillessen commented on install.sh:
```diff
@@ -1,11 +1,31 @@
+	echo -n
+}
+
+begin() {
```
Why did you extract this simple method as an additional function? Does not make it more readable.

lavirott replied:
Yes and for maintenance. If you want to add or remove any thing at each step of the script, you just have to change the begin and end function.

sebastianzillessen commented on install.sh:
```diff
@@ -1,11 +1,31 @@
+	echo -n $1
+}
+
+end() {
```
Why are you adding this pause command everywhere? Please remove it and inline the `echo "done"` again.

lavirott replied:
As I said just before, if you want to avoid displaying " done." or want to display anything else, you just have to change this function (and not to go to each inline echo to modify it).

sebastianzillessen reviewed (commented):
Hi @lavirott thanks so much for your effort. A lot changed in the `install.sh` script. 
Could you explain the individual comments a little bit more? Its hard for me to verify the changes right now as I don't have a raspberry Pi up and running.

Maybe you can explain the changes you did a little bit further.

lavirott committed:
Rollback some comments for pull request
Update corresponding to script modification
Correct one sudo issue

lavirott committed:
Replaced default AP name by hostanme (more accurate for a default)
Redirect script error messages to standard error
Suppress the sentence in French for the same one in English (and add comment)

lavirott commented:
This new pull request should correct all the issues you pointed out. Do not hesitate to ask me if you have any other question about the new code.

sebastianzillessen commented:
Great thanks. I will try to get hold of a R-Pi in the next couple of weeks and then I will test your changes there. Thanks for contributing back!
